### PR TITLE
Validate AWS machine type on cluster ig update

### LINKED ISSUE #3252: Kops should fail instantly when an unsupported instance type is used
Currently `kops create cluster` called with an unsupported instance type, like m1.medium, for --master-size/--node-size will enter an infinite loop, with the following warning:
* Nodes:
```
W0822 12:16:20.082441    7113 executor.go:109] error running task "LaunchConfiguration/nodes.foo" (4m4s remaining to succeed): instance type not handled: "m1.medium"
```
* Masters:
```
W0822 12:16:20.082483    7113 executor.go:109] error running task "LaunchConfiguration/master-eu-west-1a.masters.foo" (4m4s remaining to succeed): instance type not handled: "m1.medium"
```
* Then:
```
I0822 12:16:30.795207    7113 executor.go:124] No progress made, sleeping before retrying 2 failed task(s)
```
That gets in the way of automation.

Fail early.

PS. Version 1.6.0 (git-f69a313)

### PR DESCRIPTION
**Pull request by KashifSaadat**
Fixes #3252

### CODE REVIEW

File: pkg/apis/kops/validation/aws.go [hunk 1]
```diff
@@ -49,3 +53,15 @@ func awsValidateAdditionalSecurityGroups(fieldPath *field.Path, groups []string)
 
 	return allErrs
 }
+
+func awsValidateMachineType(fieldPath *field.Path, machineType string) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if machineType != "" {
```
Comment (chrislovecnm) [inline pkg/apis/kops/validation/aws.go]: This is great! But can we only run this when the cluster is on aws? I think this will bust GCE and vsphere ;)

Comment (chrislovecnm) [inline pkg/apis/kops/validation/aws.go]: Oh and should machine type ever == ""?

**Comment by KashifSaadat** [comment]
Hey @chrislovecnm, cheers for the feedback!

The `machineType != ""` was included due to a few test failures (`aws_test` unit tests and `create_cluster_integration_test`). The unit test files are easy to correct (just include machine type in the instance group spec that's passed to the validation function).
The integration tests all pass valid Cluster Spec config files (including machine type), but the failures appear to be on the options.yaml which I don't think is right?
```
--- FAIL: TestCreateClusterMinimal (1.29s)
	create_cluster_integration_test.go:121: error running create cluster (../../tests/integration/create_cluster/minimal/options.yaml): master-us-test-1a.spec.machineType: Invalid value: "": machine type specified is invalid
```

The `awsValidateMachineType` function is called by `awsValidateInstanceGroup`, which is only called when the cloud provider is aws: https://github.com/kubernetes/kops/blob/master/pkg/apis/kops/validation/legacy.go#L524-L530
There's currently no validation steps implemented for other cloud providers but would be good to support that in the future.

**Comment by justinsb** [comment]
I think we can add additional validation for an empty machineType later.

/lgtm

**Pull request merged**